joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/utils/NodeBuilders.scala

```diff
@@ -128,6 +128,10 @@ object NodeBuilders {
       .lineNumber(line)
   }
 
+  @deprecated(
+    "Deprecated in favour of the corresponding method io.joern.x2cpg.AstNodeBuilder and will be removed in a future version",
+    "4.0.284"
```

Comment:
I'll probably need to adjust the deprecation version, but I'll update that once I know which version this will be released in